lib/ash/data_layer/mnesia/matchspec.ex

```diff
@@ -0,0 +1,253 @@
+defmodule Ash.DataLayer.Mnesia.MatchSpec do
```

Comment:
This is really cool! I think this actually dovetails into something I'd really like to exist for this, however, which is something called `Ash.Filter.split`, which you pass in a filter, and a function that says "do you support this expression element" and it gives you back `{ands_that_you_support, ors_that_you_dont}`, that way you can use the first thing to build your own filter statement, and the second thing can be done with runtime filtering at the end.

Would you be open to this? This is *huge* BTW.

Reply:
That sounds like a cool solution. 🆒 

I think that would definitely clean up the logic a bit and then we wouldn't have to do an all or nothing approach that we are here. My only concern is, will the filters split that way naturally. Or is there fallout if we do the split and run some in the datalayer and some at runtime?

Reply:
The concept with splitting is that you only split across `*and*` boundaries. i.e

```
a == 1 and (b == 2 or has(c, "value"))
```

You'd get back

`{expr(a == 1), expr(b == 2 or has(c, "value"))}`.

So the left filter can always be applied, and the right is applied afterwards.